### PR TITLE
Document EC2 SD config

### LINKED ISSUE #221: Document EC2 service discovery

### PR DESCRIPTION
**Pull request by mjibson**
fixes #221

### CODE REVIEW

File: content/docs/operating/configuration.md [hunk 1]
```diff
@@ -164,6 +164,10 @@ serverset_sd_configs:
 file_sd_configs:
   [ - <file_sd_config> ... ]
 
+# List of EC2 service discovery configurations.
```
Comment (brian-brazil) [inline content/docs/operating/configuration.md]: This should go above file, as file and target group are kinda "in-built"

File: content/docs/operating/configuration.md [hunk 2]
```diff
@@ -456,6 +460,50 @@ names:
...
+
+# Refresh interval to re-read the instance list.
+[ refresh_interval: <duration> | default = 60s ]
+
+# The port to scrape metrics from. If using the public IP address, this must
+# instead be specified in the relabeling rule.
+[ port: <int> | default = 80 ]
+```
+
+For example, to use the public IP address on port 8080, the following relabeling may be done:
```
Comment (brian-brazil) [inline content/docs/operating/configuration.md]: This is reference documentation, so we should avoid examples here to keep it brief.

**Comment by mjibson** [comment]
Updated: moved ec2 note above file, removed example.

File: content/docs/operating/configuration.md [hunk 3]
```diff
@@ -456,6 +460,41 @@ names:
 Where `<filename_pattern>` may be a path ending in `.json`, `.yml` or `.yaml`. The last path segment
 may contain a single `*` that matches any character sequence, e.g. `my/path/tg_*.json`.
 
+### EC2 SD configurations `<ec2_sd_config>`
```
Comment (brian-brazil) [inline content/docs/operating/configuration.md]: This should be moved up one so that it's in the same order as the list up top.

**Comment by fabxc** [comment]
:+1: after Brians last comment.
@juliusv any nits?

File: content/docs/operating/configuration.md [hunk 4]
```diff
@@ -456,6 +460,41 @@ names:
...
+
+CAUTION: EC2 SD is in beta: breaking changes to configuration are still
+likely in future releases.
+
+EC2 SD configurations allow retrieving scrape targets from AWS EC2 instances. The private IP address is used by default, but my be changed to the public IP address with relabeling.
```
Comment (juliusv) [inline content/docs/operating/configuration.md]: Please wrap long lines at around ~80 chars.

**Comment by juliusv** [comment]
:+1: otherwise

**Comment by mjibson** [comment]
Updated: moved the docs above the file docs, wrapped long line at 80 chars.

**Comment by fabxc** [comment]
Did you use a WYSIWYG editor? The formatting seems to be very broken now and looks like [this](https://github.com/mjibson/docs/blob/b04c4a49338b4c3906a2adff6b1e03e6e3ced037/content/docs/operating/configuration.md#ec2-sd-configurations-ec2_sd_config).

**Comment by mjibson** [comment]
Whoops. Didn't save after an undo. Fixed.

**Comment by fabxc** [comment]
Thanks :+1:

**Pull request merged**